engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -125,31 +126,83 @@ object WorkflowActor {
 
   case object Restart extends WorkflowActorMessage with StartMode {
```

Comment:
You didn't start this behavior although this PR made it large enough for me to notice. 

I don't feel that actor message types should themselves be balls of functionality - IMO the message type should be either a simple flag to do something or maybe a bit of state as well, and then the actor itself should be working out what to do with it.